packages/flutter/lib/src/rendering/viewport.dart

```diff
@@ -14,6 +14,27 @@ enum ViewportAnchor {
   end,
 }
 
+class ViewportDimensions {
+  const ViewportDimensions({
+    this.contentSize: Size.zero,
+    this.containerSize: Size.zero
+  });
+
+  static const ViewportDimensions zero = const ViewportDimensions();
+
+  final Size contentSize;
+  final Size containerSize;
+
+  Offset getAbsolutePaintOffset({ Offset paintOffset, ViewportAnchor anchor }) {
+    switch (anchor) {
```

Comment:
and add a comment saying that if we ever change that, ViewportAnchor needs to deal with more than "start" and "end", since there's many places you could try to anchor to (e.g. center bottom, top right, etc)